Direct users to official Linux install instruction

rmloveland opened the pull request:
... not bespoke `curl` commands that can break.

Fixes DOC-13739

rmloveland commented:
@taroface i took a shot at just pointing users to the official install instructions & getting rid of the bespoke `curl` commands. PTAL!  Once we get something we're happy with, I'll backport it to our other supported versions v23.1+

rmloveland commented:
@jhlodin adding you as an FYI since this is a 'deployment & ops' change

jhlodin commented:
+1 to what Ryan said, but also the new target page doesn't actually have any instructions for downloading, extracting, and adding the binary to the PATH. In other words, this now sends people off to another page to be told to run commands that were on the original page.

I think the general idea of updating these snippets to point to the Linux install doc is on-point, but the common download/extract/copy code blocks need to move to the install doc.

rmloveland commented:
> +1 to what Ryan said, but also the new target page doesn't actually have any instructions for downloading, extracting, and adding the binary to the PATH. In other words, this now sends people off to another page to be told to run commands that were on the original page.
> 
> I think the general idea of updating these snippets to point to the Linux install doc is on-point, but the common download/extract/copy code blocks need to move to the install doc.

makes sense, i updated the Linux install page to have all the download/extract/install instructions there, and pointed everything else to that page

i also updated the way we reference `VERSION` and `ARCHITECTURE` on that page in a few places because even though it's nice we can refer to specific versions using the `{{page.version.version}}` Liquid tag, it is wrong sometimes (e.g. due to the reason that led to this PR), and we were also (incorrectly IMO) assuming the world is on x86_64 which is increasingly not true. I don't think it's a huge lift for users to know their own version/arch combo 

anyway PTAL and let me know what you think!

rmloveland commented on src/current/_includes/v25.2/prod-deployment/insecure-test-load-balancing.md:
```diff
@@ -12,12 +12,11 @@ For comprehensive guidance on benchmarking CockroachDB with TPC-C, see [Performa
     This should be a machine that is not running a CockroachDB node.
 
-1. Download the [CockroachDB archive](https://binaries.cockroachdb.com/cockroach-{{ page.release_info.version }}.linux-amd64.tgz) for Linux, and extract the binary:
+1. Download [CockroachDB for Linux]({% link {{ page.version.version }}/install-cockroachdb-linux.md %}), extract the binary, and copy it into the `PATH`:
```
ah sorry i missed a couple of spots. @taroface and @jhlodin pls let me know what you think of the approach shown in the other files of linking everything to 'Install CockroachDB on Linux' and if you are on board I'll go through and update all of these usages as well

rmloveland commented:
> To be clear, the intent is to link people to the doc that takes them down a more regularly maintained flow to find the binaries they want, right?

Yes!

> If so, it looks like the Install Linux page links out to the CockroachDB Releases page, and you kinda have to dig for a moment to find this. Would it be friendlier to just link directly to that section (substituting the page version)? eg, https://www.cockroachlabs.com/docs/releases?#v25-2

I saw Joe's comment that he was also in favor of this and I experimented with adding the `{{ page.version.version }}` to the link on the Linux install page, unfortunately that page variable yields `v25.2`, not `v25-2`. I considered writing some more Liquid to make it do that but I fear it would be a very brittle solution vs. the current link which is known good

Re: the overall PR, I just updated it so all the various things now funnel to the Linux install page as @jhlodin LGTM'd that approach in [this comment](https://github.com/cockroachdb/docs/pull/19625#discussion_r2121627479)

PTAL - If you guys are happy with this, I will then backport it to other supported versions before merge

However if you want additional changes, please let me know! Happy to fix anything needed

taroface reviewed (commented):
One comment, but otherwise LGTM.

rmloveland commented:
Thanks for the reviews @taroface and @jhlodin !

I'm going to squash and then rebase onto `main` and then port these changes to the other supported versions following the process in https://cockroachlabs.atlassian.net/wiki/x/eQDr3

taroface commented on src/current/v25.2/install-cockroachdb-linux.md:
```diff
@@ -25,10 +25,21 @@ Use one of the options below to install CockroachDB. To upgrade an existing clus
+      <p>Visit <a href="/docs/releases/index.html">Releases</a> to download the CockroachDB archive for the architecture of your Linux host. The archive contains the <code>cockroach</code> binary and the supporting libraries that are used to provide <a href="{% link {{ page.version.version }}/spatial-data-overview.md %}">spatial features</a>. Follow the steps shown below to install the <code>cock ...
+      <ol>
+        <li>
+          Download the binary you want from the <a href="/docs/releases/index.html">Releases</a> page. In the following commands, replace <code>{VERSION}</code> with the version of CockroachDB you are installing, and replace <code>{ARCHITECTURE}</code> with <code>linux-amd64</code> for Intel, or with <code>linux-arm64</code> for ARM.
```
I still think we should be more specific about the version syntax, because IMO it's extremely likely people will not prepend the `v`, at least (not sure if the liquid will still render with the code ticks around it):

```suggestion
          Download the binary you want from the <a href="/docs/releases/index.html">Releases</a> page. In the following commands, replace <code>{VERSION}</code> with the version of CockroachDB you are installing (e.g., `{{ page.version.version }}`, and replace <code>{ARCHITECTURE}</code> with <code>linux-amd64</code> for Intel, or with <code>linux-arm64</code> for ARM.
```
